jetbrains-core/src/software/aws/toolkits/jetbrains/services/apprunner/actions/ServiceUrlActions.kt

```diff
@@ -14,7 +16,19 @@ import java.awt.datatransfer.StringSelection
 
 class CopyServiceUrlAction : SingleResourceNodeAction<AppRunnerServiceNode>(message("apprunner.action.copyServiceUrl")), DumbAware {
     override fun actionPerformed(selected: AppRunnerServiceNode, e: AnActionEvent) {
-        CopyPasteManager.getInstance().setContents(StringSelection(selected.service.serviceUrl()))
+        CopyPasteManager.getInstance().setContents(StringSelection(selected.service.urlWithProtocol()))
         ApprunnerTelemetry.copyServiceUrl(selected.nodeProject)
     }
 }
+
+class OpenServiceUrlAction internal constructor(private val launcher: BrowserLauncher) :
+    SingleResourceNodeAction<AppRunnerServiceNode>(message("apprunner.action.openServiceUrl")), DumbAware {
+    constructor() : this(BrowserLauncher.instance)
```

Comment:
If you want to avoid the constructor injection stuff, you can do a runtime replacement of the BrowserLauncher service in the tests: https://github.com/aws/aws-toolkit-jetbrains/blob/412f0d93a621dd38b5f57568a9f9c947c0b72f9b/jetbrains-core/tst/software/aws/toolkits/jetbrains/utils/MockFileChooser.kt#L37

Reply:
done